ROX-11776: Integrate with NP-Guard including error handling (#2637)
stackrox/stackrox: roxctl/generate/README.md

# Static Network Policy Generator

## Developer Preview Notice

The static network policy generation feature is offered as a _developer preview_ feature.

> While we are open to receiving feedback about this feature, our technical support will be able to

Reviewer: ```suggestion
While we are open to receiving feedback about this feature, our technical support will not be able to
```

Reviewer: There is no support for DP, right?

Author: See [edfb373](https://github.com/stackrox/stackrox/pull/2637/commits/edfb373fe06d368440609f02b30e24003870cf9a)

assist and answer questions about it.

## About

> The static network policy generator a tool analyzes k8s manifests and generates a set of suggested network policies in form of yaml documents that may be directly applied to a k8s cluster. It is integrated with [NP-Guard](https://np-guard.github.io/) _analyze & synthesize_ module that discovers the network connections and generates the network policies. For more details, refer to the [NP-Guard webpage](https://np-guard.github.io/).

Reviewer: ```suggestion
The static network policy generator a tool analyzes k8s manifests and generates a set of suggested network policies in form of yaml documents that may be directly applied to a k8s cluster. It is integrated with NP-Guards _cluster-topology-analyzer_ module that discovers the network connections and generates the network policies. For more details, refer to the [NP-Guard webpage](https://np-guard.github.io/).
```
I think it is called the Cluster Topology Analyzer, but I might be wrong

Reviewer: Also, I've removed one of the NP-Guard links to not have the same link within one sentence distance two times

Author: Let me then link `Cluster Topology Analyzer` to its Github repo and keep the second link as it is.

Author: See [edfb373](https://github.com/stackrox/stackrox/pull/2637/commits/edfb373fe06d368440609f02b30e24003870cf9a)

> The static network policy generator a tool analyzes k8s manifests and generates a set of suggested network policies in form of yaml documents that may be directly applied to a k8s cluster. It is integrated with [NP-Guard's Cluster Topology Analyzer](https://github.com/np-guard/cluster-topology-analyzer) component, which discovers the network connections and generates the network policies. For more details, refer to the [NP-Guard webpage](https://np-guard.github.io/).

Reviewer 2: ```suggestion
The static network policy generator is a tool that analyzes k8s manifests and generates a set of suggested network policies in form of yaml documents that may be directly applied to a k8s cluster. It is integrated with [NP-Guard's Cluster Topology Analyzer](https://github.com/np-guard/cluster-topology-analyzer) component, which discovers the network connections and generates the network policies. For more details, refer to the [NP-Guard webpage](https://np-guard.github.io/).
```
